src/components/DataProvider/index.js

```diff
@@ -157,6 +154,37 @@ export default class DataProvider extends Component {
     if (reason !== 'UPDATE_SUBDOMAIN') {
       update.contextSeries = series;
     }
+    if (reason === 'INTERVAL') {
+      let domainMax = this.state.domain[1];
+      Object.keys(series).forEach(key => {
+        const { data, xAccessor } = series[key];
+        const serieXMax = xAccessor(data[data.length - 1]);
+        if (serieXMax > domainMax) {
+          domainMax = serieXMax;
+        }
+      });
+
+      update.domain = [this.state.domain[0], domainMax];
+
+      // follow most right subdomain
+      if (this.state.subDomain[1] === this.state.domain[1]) {
+        update.subDomain = [this.state.subDomain[0], domainMax];
+      }
+
+      // shift serie data array if a new item has been added to the end
+      if (this.props.shift) {
+        Object.keys(series).forEach(key => {
+          const newSerieLength = series[key].data.length;
+          const oldSerieLength = this.state.series[key].data.length;
+          if (newSerieLength > oldSerieLength) {
+            const lengthDiff = newSerieLength - oldSerieLength;
+            for (let i = 0; i < lengthDiff; i += 1) {
```

Comment:
suggestion: You could do `for (let i = oldSerieLength; i < newSerieLength; i += 1)` instead of calculating the diff.

If you think that's worse readability, then I'm indifferent.

Reply:
agree